apps/example-server/src/main/permissions/permissions.routes.ts

```diff
@@ -0,0 +1,76 @@
+import { and, eq, getTableColumns } from 'drizzle-orm'
+import { Hono } from 'hono'
+import { jwt } from 'hono/jwt'
+import { z } from 'zod'
+import { db } from '../../core/db/db'
+import { permissionsTable } from '../../core/db/schema'
+
+const app = new Hono()
+
+const secret = process.env.ACCESS_TOKEN_SECRET ?? ''
+
+const authMiddleware = jwt({ secret })
+
+// GET /permissions - list all
+app.get('/permissions', authMiddleware, async (c) => {
+    const permissions = await db
+        .select({ ...getTableColumns(permissionsTable) })
+        .from(permissionsTable)
+        .limit(100) // Adjust limit as needed
+
+    return c.json({ data: permissions, message: 'Permissions list' })
+})
+
+// POST /permissions - create one
+app.post('/permissions', authMiddleware, async (c) => {
+    const permissionSchema = z.object({
+        groupId: z.number(),
+        roleId: z.number(),
+        area: z.string().min(1),
+        access: z.number().min(0).max(5).optional(),
+    })
+
+    const body = await c.req.json()
+    const parsedBody = permissionSchema.parse(body)
```

Comment:
Use the schema in `zValidate` middleware. Then it will be automatically validated. Afterward, you can access the validated & typed body object using `const body = c.req.valid('json')`